drivers/sensor/adi/max30210/max30210.c

```diff
@@ -0,0 +1,689 @@
+/*
+ * Copyright (c) 2025 Analog Devices Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT adi_max30210
+
+#include <zephyr/device.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/sensor.h>
+#include "max30210.h"
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(MAX30210, CONFIG_SENSOR_LOG_LEVEL);
+
+/**
+ * @brief Generic register access function for MAX30210
+ *
+ * @param dev Device Pointer
+ * @param addr_reg Register Address
+ * @param data Data Pointer
+ * @param read Read/Write Flag
+ * @param length Number of Bytes to Read/Write
+ * @return int 0 on success, negative error code on failure
+ */
+static int max30210_reg_access(const struct device *dev, uint8_t addr_reg, uint8_t *data, bool read,
+			       uint8_t length)
+{
+
+	const struct max30210_config *config = dev->config;
+	int ret;
+
+	if (read) {
+		ret = i2c_burst_read_dt(&config->i2c, addr_reg, data, length);
+	} else {
+		ret = i2c_burst_write_dt(&config->i2c, addr_reg, data, length);
+	}
+
+	return ret;
+}
+
+/**
+ * @brief Read from MAX30210 register
+ *
+ * @param dev Device Pointer
+ * @param reg_addr Register Address
+ * @param val Value Pointer
+ * @param length Number of Bytes to Read
+ * @return int 0 on success, negative error code on failure
+ */
+int max30210_reg_read(const struct device *dev, uint8_t reg_addr, uint8_t *val, uint8_t length)
+{
+	return max30210_reg_access(dev, reg_addr, val, true, length);
+}
+
+/**
+ * @brief Single byte write to MAX30210 register
+ *
+ * @param dev Device Pointer
+ * @param reg_addr Register Address
+ * @param val Value to Write
+ * @param length Ignored
+ * @return int
+ */
+int max30210_reg_write(const struct device *dev, uint8_t reg_addr, uint8_t val)
+{
+	return max30210_reg_access(dev, reg_addr, &val, false, 1);
+}
+
+/**
+ * @brief Multiple byte write to MAX30210 register
+ *
+ * @param dev Device Pointer
+ * @param reg_addr Register Address
+ * @param val Value Pointer
+ * @param length Number of Bytes to Write
+ * @return int 0 on success, negative error code on failure
+ */
+int max30210_reg_write_multiple(const struct device *dev, uint8_t reg_addr, uint8_t *val,
+				uint8_t length)
+{
+	if (length < 2) {
+		return -EINVAL;
+	}
+	return max30210_reg_access(dev, reg_addr, val, false, length);
+}
+
+/**
+ * @brief Update specific bits in a MAX30210 register
+ *
+ * @param dev Device Pointer
+ * @param reg_addr Register Address
+ * @param mask Bit Mask
+ * @param val Value to Set
+ * @return int 0 on success, negative error code on failure
+ */
+int max30210_reg_update(const struct device *dev, uint8_t reg_addr, uint8_t mask, uint8_t val)
+{
+	uint8_t reg_val = 0;
+	int ret;
+
+	ret = max30210_reg_read(dev, reg_addr, &reg_val, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	reg_val &= ~mask;
+	reg_val |= FIELD_PREP(mask, val);
+
+	return max30210_reg_write(dev, reg_addr, reg_val);
+}
```

Comment:
These wrapper functions might make sense for a device that supports multiple transport interfaces (I2C and SPI), but for a I2C-only device they seem like overkill. Why not just use `i2c_reg_write_byte_dt()`, `i2c_reg_update_byte()` and friends directly?